src/Microsoft.Health.SqlServer/ManagedIdentityAccessTokenHandler.cs

```diff
@@ -1,29 +1,24 @@
-﻿// -------------------------------------------------------------------------------------------------
+// -------------------------------------------------------------------------------------------------
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
 // -------------------------------------------------------------------------------------------------
 
 using System.Threading;
 using System.Threading.Tasks;
-using EnsureThat;
-using Microsoft.Azure.Services.AppAuthentication;
+using Azure.Core;
+using Azure.Identity;
 
 namespace Microsoft.Health.SqlServer;
 
 public class ManagedIdentityAccessTokenHandler : IAccessTokenHandler
 {
-    private readonly AzureServiceTokenProvider _azureServiceTokenProvider;
-
-    public ManagedIdentityAccessTokenHandler(AzureServiceTokenProvider azureServiceTokenProvider)
+    /// <inheritdoc />
+    public async Task<string> GetAccessTokenAsync(string resource, CancellationToken cancellationToken)
     {
-        EnsureArg.IsNotNull(azureServiceTokenProvider, nameof(azureServiceTokenProvider));
+        DefaultAzureCredential credential = new DefaultAzureCredential();
```

Comment:
It looks like under the hood there is a new [`WorkloadIdentityCredential`](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/identity/Azure.Identity/src/Credentials/WorkloadIdentityCredential.cs) class. I would say the options here are probably:

1. Configure this default credential to only enable both `ManagedIdentityCredential` and `WorkloadIdentityCredential` OR
2. Add a new flow for the SQL library for workload identity explicitly -- like how we have a managed identity path today

Reply:
To be clear, you want me to leave the existing ManagedIdentityAccessTokenHandler as is and create a new path that uses new handler that uses (default or managedidentity credential) only when workloadidentity is being used?